modules/tinymce/src/core/main/ts/dom/TrimNode.ts

```diff
@@ -28,20 +28,25 @@ const isBookmarkNode = function (node) {
 //   <p>text 1<span></span></p><b>CHOP</b><p><span></span>text 2</p>
 // this function will then trim off empty edges and produce:
 //   <p>text 1</p><b>CHOP</b><p>text 2</p>
-const trimNode = function (dom, node) {
+const trimNode = function (dom, node, doNotTrimSpaces?: boolean) {
   let i, children = node.childNodes;
 
   if (NodeType.isElement(node) && isBookmarkNode(node)) {
     return;
   }
 
   for (i = children.length - 1; i >= 0; i--) {
-    trimNode(dom, children[i]);
+    trimNode(dom, children[i], doNotTrimSpaces);
   }
 
   if (NodeType.isDocument(node) === false) {
     // Keep non whitespace text nodes
     if (NodeType.isText(node) && node.nodeValue.length > 0) {
+      // Do not trim nodes that contain spaces.
+      if (doNotTrimSpaces) {
```

Comment:
Why did you need this? I don't like the way this has been done tbh, but I can't suggest alternatives as I can't quite figure out why this function is being called if we don't want to trim the node.

Reply:
Ahh... I think I see. It's so it trims empty child nodes, is that correct?

Reply:
Because it trims a lot of other stuff, or does things which we want it to do. I tried to do this higher up and the result was that the focus either jumped around or completely vanished in some situations. Alternatively I'd have to move much of this code up into another function to do basically this.